src/app/external-log-in/external-log-in/external-log-in.component.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  NgComponentOutlet,
+  NgIf,
+} from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Injector,
+  Input,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  NgbModal,
+  NgbModalRef,
+} from '@ng-bootstrap/ng-bootstrap';
+import {
+  TranslateModule,
+  TranslateService,
+} from '@ngx-translate/core';
+
+import { AuthService } from '../../core/auth/auth.service';
+import { AuthMethodType } from '../../core/auth/models/auth.method-type';
+import { AuthRegistrationType } from '../../core/auth/models/auth.registration-type';
+import { Registration } from '../../core/shared/registration.model';
+import { AlertComponent } from '../../shared/alert/alert.component';
+import { AlertType } from '../../shared/alert/alert-type';
+import {
+  hasValue,
+  isEmpty,
+} from '../../shared/empty.util';
+import { ThemedLogInComponent } from '../../shared/log-in/themed-log-in.component';
+import {
+  ExternalLoginTypeComponent,
+  getExternalLoginConfirmationType,
+} from '../decorators/external-log-in.methods-decorator';
+import { ConfirmEmailComponent } from '../email-confirmation/confirm-email/confirm-email.component';
+import { ProvideEmailComponent } from '../email-confirmation/provide-email/provide-email.component';
+
+@Component({
+  selector: 'ds-external-log-in',
+  templateUrl: './external-log-in.component.html',
+  styleUrls: ['./external-log-in.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ProvideEmailComponent,
+    AlertComponent,
+    TranslateModule,
+    ConfirmEmailComponent,
+    ThemedLogInComponent,
+    NgIf,
+    NgComponentOutlet,
+  ],
+  standalone: true,
+})
+export class ExternalLogInComponent implements OnInit, OnDestroy  {
```

Comment:
Again, it'd be good to give a description of the purpose of this component as a code comment / typedocs.